src/main/java/org/isf/utils/sms/SetupGSM.java

```diff
@@ -40,74 +39,70 @@
-	
+
 	/**
 	 * @param args
 	 */
 	public static void main(String[] args) {
 		new SetupGSM();
 		System.exit(0);
 	}
-	
+
 	public SetupGSM() {
 		props = ConfigurationProperties.loadPropertiesFile(GSMParameters.FILE_PROPERTIES, LOGGER);
-		
+
 		String model = props.getProperty(GSMGatewayService.SERVICE_NAME + ".gmm");
 
-		Enumeration<?> portList = CommPortIdentifier.getPortIdentifiers();
-		
-		while (portList.hasMoreElements()) {
-			
-			portId = (CommPortIdentifier) portList.nextElement();
-
-			if (portId.getPortType() == CommPortIdentifier.PORT_SERIAL) {
-
-				System.out.println("Port found: " + portId.getName() + ' ' + (portId.getPortType() == CommPortIdentifier.PORT_SERIAL ? "SERIAL" : "PARALLEL"));
-			
-				try {
-					serialPort = (SerialPort) portId.open("SmsSender", 10);
-					serialPort.setSerialPortParams(9600, SerialPort.DATABITS_8, SerialPort.STOPBITS_1, SerialPort.PARITY_NONE);
-					serialPort.addEventListener(this);
-			        serialPort.notifyOnDataAvailable(true);
-					
-					OutputStream outputStream = serialPort.getOutputStream();
-					if (outputStream != null) {
-						System.out.println("Output stream OK");
-					} else {
-						System.out.println("Output stream not found");
-					}
-					
-					inputStream = serialPort.getInputStream(); 
-					byte[] command = model.getBytes();
-			        outputStream.write(command);
-			        
-			        Thread.sleep(5000);
-			        
-				} catch (PortInUseException e) {
-					LOGGER.error("Port in use.");
-				} catch (Exception exception) {
-					LOGGER.error("Failed to open port '{}'", portId.getName());
-					LOGGER.error(exception.getMessage(), exception);
-				} finally {
-					serialPort.close();
+		SerialPort[] portList = SerialPort.getCommPorts();
+		System.out.println("Found " + portList.length + " ports: " + portList.toString());
+
+		for (SerialPort comPort : portList) {
+
+			// System.out.println("Port found: " + portId.getName() + ' ' + (portId.getPortType() == CommPortIdentifier.PORT_SERIAL ? "SERIAL" : "PARALLEL"));
+
+			try {
+				System.out.println("Opening port...");
+				comPort.openPort();
+				comPort.setComPortTimeouts(SerialPort.TIMEOUT_READ_SEMI_BLOCKING, 0, 0);
+				comPort.addDataListener(this);
+
+				OutputStream outputStream = comPort.getOutputStream();
+				if (outputStream != null) {
+					System.out.println("Output stream OK");
+				} else {
+					System.out.println("Output stream not found");
 				}
+
+				inputStream = comPort.getInputStream();
+				byte[] command = model.getBytes();
+				outputStream.write(command);
+
+				Thread.sleep(5000);
+
+			} catch (PortInUseException e) {
+				LOGGER.error("Port in use.");
+			} catch (RuntimeException re) {
+				LOGGER.error("Something strange happened.", re);
+			} catch (Exception exception) {
+				LOGGER.error("Failed to open port '{}'", comPort);
+				LOGGER.error(exception.getMessage(), exception);
+			} finally {
+				System.out.println("Closing port...");
```

Comment:
It looks like LOGGER is already there; see lines 101 and 102.

Reply:
with the new scripts the utility is indeed called with the full classpath, so also bundles would be available...
otherwise (if no .properties are provided aside the libs) it just returns this and nothing is shown out of `println`
```
log4j:WARN No appenders could be found for logger (org.isf.utils.sms.SetupGSM).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
```
wondering if to move everything to logger or back to println though... 🤔

Output in console with LOGGER (to be improved):
```
[07/dic/2023 00:55:36] [:] INFO - File sms.properties loaded.
[07/dic/2023 00:55:36] [:] INFO - Found 1 ports: [Lcom.fazecast.jSerialComm.SerialPort;@133e16fd
[07/dic/2023 00:55:36] [:] INFO - Port found: COM3 (Intel(R) Active Management Technology - SOL (COM3))
[07/dic/2023 00:55:36] [:] INFO - Opening port...
[07/dic/2023 00:55:36] [:] INFO - Output stream OK
[07/dic/2023 00:55:41] [:] INFO - Closing port...
[07/dic/2023 00:55:41] [:] INFO - End.
Done!
```

without LOGGER:
```
Found 1 ports: [Lcom.fazecast.jSerialComm.SerialPort;@133e16fd
Port found: COM3 (Intel(R) Active Management Technology - SOL (COM3))
Opening port...
Output stream OK
Closing port...
End.
Done!
```